parsons/bluelink/person.py

```diff
@@ -0,0 +1,242 @@
+        addresses: list[Address]
+            A list of Address objects representing postal addresses
+        tags: list[Tag]
+            Simple tags that apply to the person, eg DONOR.
+        employer: str
+            Name of the persons employer
+        employer_address: Address
+            Address of the persons employer
+        occupation: str
+            Occupation
+        birthdate: str
+            ISO 8601 formatted birth date: YYYY-MM-DD
+        scores: list[Score]
+            List of Score objects. Scores are numeric scores, ie partisanship model.
+        details: dict
+            additional custom data. must be json serializable
+    """
+    def __init__(self, identifiers, given_name=None, family_name=None, phones=None, emails=None,
+                 addresses=None, tags=None, flags=None, employer=None, employer_address=None,
+                 occupation=None, scores=None, birthdate=None, details=None):
+
+        if not identifiers:
+            raise Exception("Person requires list of Identifiers with at least 1 Identifier")
+
+        self.identifiers = identifiers
+        self.addresses = addresses
+        self.emails = emails
+        self.phones = phones
+        self.tags = tags
+        self.flags = flags
+        self.scores = scores
+
+        self.given_name = given_name
+        self.family_name = family_name
+
+        self.employer = employer
+        self.employer_address = employer_address
+        self.occupation = occupation
+        self.birthdate = birthdate  # TODO: verify is ISO
+        self.details = details
+
+    def __json__(self):
+        """The json str representation of this Person object"""
+        return json.dumps(self, default=lambda obj: obj.__dict__)
+
+    def __eq__(self, other):
+        """A quick and dirty equality check"""
+        dself = json.loads(self.__json__())
+        dother = json.loads(other.__json__())
+        return dself == dother
+
+    def __repr__(self):
+        return self.__json__()
+
+    @staticmethod
+    def from_table(tbl, dict_to_person):
+        """
+        Return a list of Person objects from a Parsons Table.
+
+        `Args:`
+            tbl: Table
+                A parsons Table
+            dict_to_person: Callable[[dict],Person]
+                A function that takes a dictionary representation of a table row,
+                and returns a Person.
+        `Returns:`
+           list[Person]
+           A list of Person objects
+        """
+        return [dict_to_person(row) for row in tbl]
+
+
+class Identifier(object):
+    """
+    Instantiate an Identifier object
+    Identifier is necessary for updating Person records
+
+    `Args:`
+        source: str
+            External system to which this ID belongs, e.g., “VAN:myCampaign”.
+            Bluelink has standardized strings for source. Using these will allow
+            Lightrail to correctly understand the external IDs you add. source
```

Comment:
We're not using the name "Lightrail" externally anymore. Can just change to "Bluelink"

Reply:
Oh! I didn't know that.